zipline/pipeline/factors/technical.py

```diff
@@ -144,6 +148,205 @@ def compute(self, today, assets, out, close, volume):
         out[:] = nanmean(close * volume, axis=0)
 
 
+class CorrelationFactor(SingleInputMixin, CustomFactor):
```

Comment:
I'd also be hesitant to just call this `Correlation`, since there are several mathematical formulae that have some claim to that name.  I think what we're actually calculating here is the [Pearson Product-Moment Correlation](https://statistics.laerd.com/statistical-guides/pearson-correlation-coefficient-statistical-guide.php).  I think `scipy.stats` calls this `pearsonr`.  It also cautions that it should only be used on normally-distributed data, which I don't think is true of most of our inputs.  @CaptainKanuk thoughts on names for this?  I would lean toward something like `PearsonCorrelationCoefficient`.